pkg/autodiscover/autodiscover_scaleObject.go

```diff
@@ -0,0 +1,78 @@
+package autodiscover
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/cnf-certification-test/internal/clientsholder"
+	"github.com/test-network-function/cnf-certification-test/pkg/configuration"
+	scalingv1 "k8s.io/api/autoscaling/v1"
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type Scaleobject struct {
+	Scale               *scalingv1.Scale
+	GroupResourceSchema schema.GroupResource
+}
+
+//nolint:funlen
+func GetscaleCrUndetTest(namespaces []string, crds []*apiextv1.CustomResourceDefinition, testData []configuration.CrdFilter) []Scaleobject {
+	var scalableItems []Scaleobject
+	clients := clientsholder.GetClientsHolder()
+	for _, aCrd := range crds {
+		scalable := false
+		for _, crdFilter := range testData {
+			if crdFilter.NameSuffix == aCrd.Name {
+				scalable = crdFilter.Scalable
+				break
+			}
+		}
+		if scalable {
```

Comment:
This whole `scaleable` block could probably be moved inside the `if crdFilter.NameSuffix == aCrd.Name` check above.  The `break` could be removed as well then.